File: app/src/main/java/com/gee12/mytetroid/model/ITetroidObject.java
package com.gee12.mytetroid.model;

/**
 * Общий интерфейс для объектов всех типов в хранилище.
 */
public interface ITetroidObject {

    /**
     * Тип объекта.
     * @return
     */
    int getType();

    /**
     * Заголовок объекта.
     * @return
     */
    String getName();

    /**
     * ID объекта.
     * @return
     */
    String getId();

    /**
     * Зашифрован?
     * @return
     */
    boolean isCrypted();
}

File: app/src/main/java/com/gee12/mytetroid/model/ITetroidObject.kt
package com.gee12.mytetroid.model

/**
 * Общий интерфейс для объектов всех типов в хранилище.
 */
interface ITetroidObject {
    val type: Int
    var name: String
    val id: String
    val isCrypted: Boolean
}
